Update reqwest requirement from 0.9.0 to 0.10.1

internetionals opened the pull request:
I'm using your fantastic library in one my small tools and whilst updating and reviewing my dependencies it turned out to be the only dependency that's still pulling in tokio 0.1. And update to reqwest would alleviate that.

The other pull request triggered by dependabot fails because the blocking API has been moved to a separate module and requires a feature flag to be enabled. Apart from those changes everything appears to be working.

internetionals committed:
Update reqwest requirement from 0.9.0 to 0.10.1

jonas-schievink commented:
Awesome, thanks! Looks like we need to update the MSRV since it no longer builds on Rust 1.36.0. Can you do that? (find the oldest Rust version where this still builds and then update the version in `.travis.yml`)

internetionals committed:
Bump lowest tested compiler version to 1.37

internetionals committed:
Drop deprecated functions from std::error::Error impls

internetionals commented:
I updated the Rust version to 1.37 (still more than 3 stable releases back) and that should fix that problem.

I also dropped the soft deprecated 'fn description' from the std::error::Error impls, you already had the similar Display impls in place. But if you don't want that part I can drop it easily.

internetionals commented:
Damn, from the looks of it the fix for that error would bump the MSRV to 1.39, so that would mean it has to wait another Rust release.

jonas-schievink commented:
Ah, fair. 1.41 comes out in ~a week though, so that's not too bad.

internetionals committed:
Bump minimum tested Rust version

This is the first version with `core::slice::<impl [T]>::len` as a const fn

internetionals commented:
I've checked locally and Rust 1.39 is indeed the first stable version to be able to compile all the reqwest 0.10.1 related dependencies.

jonas-schievink commented:
Looks good, thanks! I'll publish a new version once Rust 1.41 is out (feel free to ping me if I forget).
